### ISSUE TITLE
[Automation API] Add structured errors for automation API

### ISSUE THREAD

**Issue by yarinm** [pulumi/pulumi#6137]
For our product, I need to be able to determine from the pulumi automation API errors if we have any permissions errors for running the stack and if we reach certain quotas in the relevant service (CPU limit, VPC quota etc)

Currently it's really hard to parse errors from the automation API. I saw this [ticket](https://github.com/pulumi/pulumi/issues/5218) and I was wondering if you ever discussed about having structured errors as well.

**Comment by yarinm** [comment]
@komalali @EvanBoyle is there any plans on doing that? At the moment I have to resort to running regexes on the entire pulumi output to handle certain errors.

**Comment by EvanBoyle** [comment]
> At the moment I have to resort to running regexes on the entire pulumi output to handle certain errors.
Can you provide the errors or regexes you are running? There may be some that we can move into the errors library itself: 

https://github.com/pulumi/pulumi/blob/sdk/v3.11.0/sdk/go/auto/errors.go#L44-L76
